Code/GraphMol/Descriptors/BCUT.cpp

```diff
@@ -0,0 +1,97 @@
+//
+//  Copyright (C) 2020 Brian P. Kelley
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include "BCUT.h"
+#include <Eigen/Dense>
+#include <GraphMol/RDKitBase.h>
+#include "GraphMol/PartialCharges/GasteigerCharges.h"
+#include "GraphMol/PartialCharges/GasteigerParams.h"
+#include <RDGeneral/types.h>
+
+namespace RDKit {
+namespace Descriptors {
+  // diagonal elements are a property (atomic num, charge, etc)
+  // off diagonal are .1,.2,.3.15 for single,double,triple or aromatic
+  // all other elements are .001
+  std::pair<double,double> BCUT2D(const ROMol &m, const std::string &atom_double_prop) {
+    unsigned int num_atoms = m.getNumAtoms();
+    if (num_atoms == 0) {
+        return std::pair<double,double>(0,0);
+    }
+    
+    Eigen::MatrixXd burden(num_atoms, num_atoms);
+
+    // All non bonded entires set to 0.001
+    for(unsigned int i=0;i<num_atoms;++i) {
+      burden(i,i) = m.getAtomWithIdx(i)->getProp<double>(atom_double_prop);
+      for(unsigned int j=0;j<num_atoms;++j) {
+	if (i!=j)
```

Comment:
we're putting `{}` around single line if statements too now (it's not a clang-format operation, so this needs to be done by hand).